docs/book/resource_layer.rst

```diff
@@ -0,0 +1,187 @@
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction()
+    {
+        $factory = $this->container->get('sylius.factory.product');
+
+        /** @var ProductInterface $product **/
+        $product = $factory->createNew();
+    }
+
+.. note::
+
+    Creating resources via this factory method makes the code more testable, and allows you to change the model class easily.
+
+Manager
+~~~~~~~
+
+The manager service is just an alias to appropriate Doctrine's `ObjectManager`_ and can be accessed via the *sylius.manager.product* id.
+API is exactly the same and you are probably already familiar with it:
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction()
+    {
+        $manager = $this->container->get('sylius.manager.product');
+
+        // Assuming that the $product1 exists in the database we can perform such operations:
+        $manager->remove($product1);
+
+        // If we have created the $product2 using a factory, we can persist it in the database.
+        $manager->persist($product2);
+
+        // Before performing a flush, the changes we have made, are not saved. There is only the $product1 in the database.
+        $manager->flush(); // Saves changes in the database.
+
+        //After these operations we have only $product2 in the database. The $product1 has been removed.
+    }
+
+Repository
+~~~~~~~~~~
+
+Repository is defined as a service for every resource and shares the API with standard Doctrine *ObjectRepository*. It contains two additional methods for creating a new object instance and a paginator provider.
+
+The repository service is available via the *sylius.repository.product* id and can be used like all the repositories you have seen before.
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction()
+    {
+        $repository = $this->container->get('sylius.repository.product');
+
+        $product = $repository->find(4); // Get product with id 4, returns null if not found.
+        $product = $repository->findOneBy(['slug' => 'my-super-product']); // Get one product by defined criteria.
+
+        $products = $repository->findAll(); // Load all the products!
+        $products = $repository->findBy(['special' => true]); // Find products matching some custom criteria.
+    }
+
+Every Sylius repository supports paginating resources. To create a `Pagerfanta instance <https://github.com/whiteoctober/Pagerfanta>`_ use the ``createPaginator`` method.
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction(Request $request)
+    {
+        $repository = $this->container->get('sylius.repository.product');
+
+        $products = $repository->createPaginator();
+        $products->setMaxPerPage(3);
+        $products->setCurrentPage($request->query->get('page', 1));
+
+        // Now you can return products to template and iterate over it to get products from current page.
+    }
+
+Paginator can be created for a specific criteria and with desired sorting.
```

Comment:
Should this and the previous two sentances before code blocks be terminated with `:` ?